docs/setup/k8s/running-in-production-k8s.md

```diff
@@ -0,0 +1,122 @@
+# Running Nuclio over Kubernetes in production
+
+After familiarizing yourself with Nuclio, and [deploying it over k8s](/docs/setup/k8s/getting-started-k8s.md) you may find yourself looking for extra configuration knobs, and proper practices for using it in a production environment.
+In [Iguazio](https://www.iguazio.com/) we integrated Nuclio as part of our [Data science PaaS](https://www.iguazio.com/platform/), and it is in extensive use in production, for both our customers and ourselves, running various workloads.
+
+Here you will find more advanced configuration options, and some practices which address the needs of running Nuclio in production environments. 
+
+
+#### In this document
+
+- [Preferred installation method](#preferred-installation-method)
+- [Version freezing](#version-freezing)
+- [Multi Tenancy](#multi-tenancy)
+- [Air gapped (dark site) operation](#air-gapped-dark-site-operation)
+- [Using Kaniko as an image builder](#using-kaniko-as-an-image-builder)
+
+
+## Preferred installation method
+
+Even though you may, of course, use any of the available methods, we find that heavy-lifting over k8s is best done with helm charts.
+Nuclio's [helm chart](/hack/k8s/helm/nuclio/) is our preferred mode of deploying Nuclio in [Iguazio](https://www.iguazio.com/) as of writing this document.
+This is also where you'll find most of the production oriented features appearing first, and is the most tightly maintained form of deployment.
+
+Below is a quick example of how to setup the a specific stable version of nuclio (1.3.14) via helm charts:
+
+- Create a namespace for your functions, and a secret with your intended registry's credentials
+
+    ```sh
+    kubectl create namespace nuclio
```

Comment:
Is it really needed?
Why not simply use default namespace?

Reply:
In all examples in the documentation, the nuclio functions are created in a dedicated nuclio namespace (barring multi tenancy scenarios, this is the simplest one).
Since it's in more basic examples, like in the getting started, I don't see a reason to regress  in this doc